### PR TITLE
[examples] Add cmake for ResNet50 bundle

### PR DESCRIPTION
**Pull request by SplitInfinity**
**Description**
This commit adds the capability to build the ResNet50 standalone bundle
using `cmake` in addition to `make`. Quantization is controlled for all
bundles using the `GLOW_QUANTIZE_BUNDLES` `cmake` variable.

**Testing**
Configured fresh build and ran bundle running cmake target:

```
$ ninja RunResNet50Bundle -v
...
...
...
Allocated weights of size: 102121984
Expected weights of size: 102121984
Loaded weights of size: 102121984 from the file resnet50.weights
Allocated mutable weight variables of size: 606144
Loaded image: {...}/cat_285.png
Loaded image: {...}/dog_207.png
Loaded image: {...}/zebra_340.png
Loaded images size in bytes is: 1806336
Copying image data into mutable weight vars: 1806336 bytes
Result: 285
```

**Fixes**
Part of issue #1979.

### CODE REVIEW

**Comment by SplitInfinity** [comment]
Travis fails because I'm using a feature from `cmake 3.11+`. Below that version, you're not allowed to use `add_executable` without providing any sources. I more or less translated the Makefile for the ResNet bundle, so I used this feature for the executable target that takes in the driver and bundle object files as input and produces the final executable.

File: examples/bundles/resnet50/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    RunBundleCommand
+  COMMAND
+    ResNet50Bundle ${IMAGES}/*.png
+  DEPENDS
+    ResNet50Bundle
+)
+add_custom_target(RunResNet50Bundle DEPENDS RunBundleCommand ResNet50Bundle)
+
+# Final executable
+add_executable(ResNet50Bundle)
```
Comment (bertmaher) [inline examples/bundles/resnet50/CMakeLists.txt]: Would it work to declare this as
```
add_executable(ResNet50Bundle $<TARGET_OBJECTS:ResNet50BundleMain>)
```

**Comment by bertmaher** [comment]
This is really nice work, thanks for digging into CMake and figuring this out.  It's pretty magical to have the bundle rebuild automatically when glow changes, and not have to manually supply GLOW_SRC and other environment variables.

**Comment by bertmaher** [comment]
How would you feel about making separate targets for quantized and non-quantized bundles, rather than conditionally compiling based on a config parameter?  I could see wanting to build/test both without reconfiguring my build tree.

**Comment by SplitInfinity** [comment]
Hmm, after separating out the quantized and non-quantized targets and taking @bertmaher 's suggestion on how to make the commit compatible with `cmake` 3.6, Travis fails with this:

```
[114/171] Generating resnet50.o
FAILED: cd /home/travis/build/pytorch/glow/build/examples/bundles/resnet50 && /home/travis/build/pytorch/glow/build/bin/image-classifier /home/travis/build/pytorch/glow/tests/images/imagenet/dog_207.png -g -image_mode=0to1 -m=/home/travis/build/pytorch/glow/build/examples/bundles/resnet50/resnet50 -model_input_name=gpu_0/data -cpu -emit-bundle /home/travis/build/pytorch/glow/build/examples/bundles/resnet50
image-classifier: Unknown command line argument '-g'.  Try: '/home/travis/build/pytorch/glow/build/bin/image-classifier -help'
image-classifier: Did you mean '-i'?
```

`image-classifier` definitely has a `-g` flag. Has anyone seen this before?

**Comment by SplitInfinity** [comment]
It turns out `-g` is a `CPUBackend` flag and Travis compiles with that disabled. I've gated the bundle builds on `GLOW_WITH_CPU` to address this.

**Comment by SplitInfinity** [comment]
Okay, I think this PR is ready for review. In particular, I'd like feedback on:
- CMakefile style and convention (if only `clang-format` took care of these too!)
- mode-specific build instructions? (e.g. do not allow quantization in DEBUG mode because it takes forever)

Once everything looks good, I'll migrate the other bundles either in this PR or a separate PR depending on what everyone prefers.

**Comment by rdzhabarov** [comment]
Let's disable bundles by default it takes 40mins to run on CI: https://github.com/pytorch/glow/pull/2021

**Pull request merged**